testing-internal/src/main/java/com/linecorp/armeria/testing/internal/MockAddressResolverGroup.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.testing.internal;
+
+import static java.util.Objects.requireNonNull;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.util.Exceptions;
+
+import io.netty.resolver.AddressResolver;
+import io.netty.resolver.AddressResolverGroup;
+import io.netty.resolver.InetNameResolver;
+import io.netty.resolver.InetSocketAddressResolver;
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.Promise;
+
+/**
+ * An {@link AddressResolverGroup} which returns an IP address resolved by the resolver function.
+ */
+public final class MockAddressResolverGroup extends AddressResolverGroup<InetSocketAddress> {
+
+    private final Function<String, InetAddress> resolver;
+
+    /**
+     * Creates an {@link AddressResolverGroup} which always returns {@code 127.0.0.1} for any hostname.
+     */
+    public MockAddressResolverGroup() {
```

Comment:
How about making the constructors private and adding static factory methods so that a user knows what it resolves to? e.g. `MockAddressResolverGroup.localhost()`, `always(String ip)` and `of(Function)`

Reply:
Good suggestion. :-)